driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -452,14 +456,15 @@ private void notifyOwnerWhenDefunct() {
     }
 
     String keyspace() {
-        return keyspace;
+        SetKeyspaceAttempt currentAttempt = targetKeyspace.get();
+        return currentAttempt == null ? null : currentAttempt.keyspace;
```

Comment:
can get around this by setting it to the same thing set in the Connection constructor, will do that.